akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -276,18 +318,21 @@ private[stream] object ConnectionSourceStage {
       override def onPush(): Unit = {
         val elem = grab(bytesIn)
         ReactiveStreamsCompliance.requireNonNullElement(elem)
-        connection ! Write(elem.asInstanceOf[ByteString], WriteAck)
-      }
+        if (writePending) {
```

Comment:
this flag seems to be more like `writeAckPending`? Consider rename or my reading is weird?

Reply:
you are right, that is a better name, and perhaps `writeInProgress` might be even better

Reply:
Yeah, that's better :)